recordedit/form.controller.js

```diff
@@ -141,58 +145,143 @@
+        }
+
+        function uploadFiles(submissionRowsCopy, isUpdate, onSuccess) {
+
+            // If url is valid
+            if (areFilesValid(vm.recordEditModel.submissionRows)) {
+
+                $uibModal.open({
+                    templateUrl: "../common/templates/uploadProgress.modal.html",
+                    controller: "UploadModalDialogController",
+                    controllerAs: "ctrl",
+                    size: "md",
+                    backdrop: 'static',
+                    keyboard: false,
+                    resolve: {
+                        params: {
+                            reference: $rootScope.reference,
+                            rows: submissionRowsCopy
+                        }
+                    }
+                }).result.then(onSuccess, function(exception) {
+                    vm.readyToSubmit = false;
+                    vm.submissionButtonDisabled = false;
+                    AlertsService.addAlert(exception.message, 'error');
+                });
+            } else {
                 vm.readyToSubmit = false;
-                AlertsService.addAlert('Sorry, the data could not be submitted because there are errors on the form. Please check all fields and try again.', 'error');
-                form.$setSubmitted();
-                return;
+                vm.submissionButtonDisabled = false;
             }
+        }
 
-            // Form data is valid, time to transform row values for submission to ERMrest
-            vm.readyToSubmit = true;
-            vm.submissionButtonDisabled = true;
-            for (var j = 0; j < model.rows.length; j++) {
-                // in the copy case, there will only ever be one tuple. Each additional form should be based off of the original tuple
-                if (vm.editMode) {
-                    originalTuple = $rootScope.tuples[j];
-                }else if (context.queryParams.copy) {
-                    originalTuple = $rootScope.tuples[0];
-                } else {
-                    originalTuple = null;
-                    editOrCopy = false;
-                }
-                populateSubmissionRow(model.rows[j], model.submissionRows[j], originalTuple, $rootScope.reference.columns, editOrCopy);
-            }
+        function addRecords(isUpdate) {
+            var model = vm.recordEditModel;
+            var form = vm.formContainer;
 
-            if (vm.editMode) {
-                // loop through model.submissionRows
-                for (var i = 0; i < model.submissionRows.length; i++) {
-                    var row = model.submissionRows[i];
-                    var data = $rootScope.tuples[i].data;
-                    // assign each value from the form to the data object on tuple
-                    for (var key in row) {
-                        data[key] = (row[key] === '' ? null : row[key]);
+            var submissionRowsCopy = [];
+
+            model.submissionRows.forEach(function(row) {
```

Comment:
I think this was something that we needed to change. The previous code was modifying data in place. Thank you for addressing this.